tests/avify4mtest.c

```diff
@@ -0,0 +1,177 @@
+        if (image->yuvPlanes[plane] != NULL) {
+            const uint32_t planeWidth = (plane == AVIF_CHAN_Y) ? image->width
+                                                               : ((image->width + info.chromaShiftX) >> info.chromaShiftX);
+            const uint32_t planeHeight = (plane == AVIF_CHAN_Y) ? image->height
+                                                                : ((image->height + info.chromaShiftY) >> info.chromaShiftY);
+            for (uint32_t y = 0; y < planeHeight; ++y) {
+                uint8_t * const row = image->yuvPlanes[plane] + y * image->yuvRowBytes[plane];
+                if (image->depth == 8) {
+                    memset(row, yuvValue, planeWidth);
+                } else {
+                    for (uint32_t x = 0; x < planeWidth; ++x) {
+                        ((uint16_t *)row)[x] = yuvValue;
+                    }
+                }
+            }
+        }
+    }
+    if (image->alphaPlane != NULL) {
+        const uint32_t alphaValue = ((image->alphaRange == AVIF_RANGE_LIMITED) ? 235 : 255) << (image->depth - 8);
+        for (uint32_t y = 0; y < image->height; ++y) {
+            uint8_t * const row = image->alphaPlane + y * image->alphaRowBytes;
+            if (image->depth == 8) {
+                memset(row, alphaValue, image->width);
+            } else {
+                for (uint32_t x = 0; x < image->width; ++x) {
+                    ((uint16_t *)row)[x] = alphaValue;
+                }
+            }
+        }
+    }
+}
+
+// Creates an image and encodes then decodes it as a y4m file.
+static avifBool encodeDecodeY4m(uint32_t width, uint32_t height, uint32_t depth, avifPixelFormat yuvFormat, avifRange yuvRange, avifBool createAlpha)
+{
+    avifBool success = AVIF_FALSE;
+    avifImage * image = avifImageCreateEmpty();
+    avifImage * decoded = avifImageCreateEmpty();
+    if (image == NULL || decoded == NULL) {
+        printf("ERROR: avifImageCreate() failed\n");
+        goto cleanup;
+    }
+    image->width = width;
+    image->height = height;
+    image->depth = depth;
+    image->yuvFormat = yuvFormat;
+    image->yuvRange = yuvRange;
+    image->alphaRange = AVIF_RANGE_FULL; // Unused by y4mRead() and y4mWrite().
+    avifImageAllocatePlanes(image, createAlpha ? AVIF_PLANES_ALL : AVIF_PLANES_YUV);
+    fillPlanes(image);
+
+    if (!y4mWrite(kPath, image)) {
+        printf("ERROR: y4mWrite() failed\n");
+        goto cleanup;
+    }
+    if (!y4mRead(kPath, decoded, /*sourceTiming=*/NULL, /*iter=*/NULL)) {
+        printf("ERROR: y4mRead() failed\n");
+        goto cleanup;
+    }
+
+    if (!compareYUVA(image, decoded)) {
+        goto cleanup;
+    }
+
+    success = AVIF_TRUE;
+cleanup:
+    if (image != NULL) {
+        avifImageDestroy(image);
+    }
+    return success;
+}
+
+//------------------------------------------------------------------------------
+
+int main(int argc, char * argv[])
+{
+    (void)argc, (void)argv;
```

Comment:
We can declare the main() function as follows:

`int main(void)`

Does the compiler warn if we don't use `argc` and `argv`? I wonder if the compiler makes an exception for the `argc` and `argv` parameters of `main()`.

Reply:
I got a compiler error about unused variables, yes. I tried anonymous arguments but this is not valid in C.